src/plugins/chromecastPlayer/plugin.js

```diff
@@ -335,16 +333,26 @@ class CastPlayer {
             apiClient = ServerConnections.currentApiClient();
         }
 
+        /* If serverAddress is localhost,this address can not be used for the cast receiver device.
+         * Use the local address (ULA, Unique Local Address) in that case.
+	 */
+        const srvAddress = apiClient.serverAddress();
+        const prefix = 'http' + ':' + '//';
+        const checkLocalhost = srvAddress.startsWith(prefix + 'localhost') || srvAddress.startsWith(prefix + '127.') || srvAddress.startsWith(prefix + '[::1]');
```

Comment:
Sorry for the delay! This is an interesting issue that you found. I would suggest using the URL API to handle some of the parsing for this.
```suggestion
        const serverAddress = apiClient.serverAddress();
        const hostname = (new URL(serverAddress)).hostname;
        const isLocalhost = hostname === 'localhost' || hostname.startsWith('127.') || hostname === '[::1]';
```

You may need to ignore an eslint warning for compatibility when using the URL constructor, but it's polyfilled by core-js so it should be fine to use.